lib/db2i.js

```diff
@@ -95,283 +94,149 @@ DB2.prototype.create = function(model, data, options, callback) {
+          }
+          self.commit(conn, function(err) {});
+          callback(null, {count: retVal});
+        });
       }
-      executeWithConnection(conn, function(err, data, meta) {
-        if (err) {
-          conn.rollbackTransaction(function(err) {
-            conn.close(function() {});
-            return callback && callback(err);
-          });
-        } else {
-          options.transaction = undefined;
-          conn.commitTransaction(function(err) {
-            if (err) {
-              return callback && callback(err);
-            }
-
-            conn.close(function() {});
-            return callback && callback(null, data, meta);
-          });
-        }
-      });
     });
   }
 };
 
-DB2.prototype.buildReplace = function(model, where, data, options) {
-  process.nextTick(function() {
-    throw Error(g.f('Function {{buildReplace}} not supported'));
-  });
-};
-
 /**
- * Update if the model based on the Where clause
+ * Build the `DELETE FROM` SQL statement
  *
  * @param {string} model The model name
- * @param {Object} where The where clause contents
- * @param {Object} data The model instance data
- * @param {Object} options to be used to check if a transaction is needed.
- * @param {Function} [callback] The callback function
+ * @param {Object} where The where object
+ * @param {Object} options Options object
+ * @param {Function} callback function
  */
-DB2.prototype.update = function(model, where, data, options, callback) {
-  debug('DB2.prototype.update (enter): ' +
-    'model=%j, where=%j, data=%j options=%j\n',
-    model, where, data, options);
+DB2iSeries.prototype.destroyAll = function(
+  model, where, options, callback) {
+  debug('DB2iSeries.prototype.destroyAll %j %j %j', model, where, options);
   var self = this;
-  var id = self.idName(model);
-  var stmt;
   var tableName = self.tableEscaped(model);
-  var meta = {};
-
-  function executeWithConnection(connection, cb) {
-    // Execution for update requires running two
-    // separate SQL statements.  The second depends on the
-    // result of the first.
-    var countStmt = new ParameterizedSQL('SELECT COUNT(*) AS CNT FROM ');
-    countStmt.merge(tableName);
-    countStmt.merge(self.buildWhere(model, where));
-    countStmt.noResults = false;
-
-    connection.query(countStmt, function(err, countData) {
-      debug('DB2.prototype.update (data): sql %j, countData=%j\n',
-            countStmt, countData);
-
+  var id = self.idName(model);
+  var deleteStmt = self.buildDelete(model, where, options);
+  var selectStmt = new ParameterizedSQL('SELECT ' + self.escapeName(id) +
+                                        ' FROM ' + tableName);
+  selectStmt.merge(self.buildWhere(model, where));
+  self.parameterize(selectStmt);
+
+  var executeTransaction = function(connection, cb) {
+    connection.query(selectStmt, function(err, selectData) {
+      debug('DB2iSeries.prototype.destroyAll stmt: %j data: %j', selectStmt,
```

Comment:
The selectData happens to contain the Error array from the statement executed so having this hear gives me the benefit of being able to dump the error content in the debug statement.